R/zchunk_LA132.Industry_china.R

```diff
@@ -0,0 +1,243 @@
+    # Industrial sector cogeneration input energy by province and fuel
+    L132.in_pct_province_ind_F %>%
+      filter(fuel %in% unique(L123.in_EJ_R_indchp_F_Yh$fuel)) %>%
+      # We only want fuels that are inputs to cogen systems, i.e. not electricity
+      mutate(sector = "chp_elec") %>%
+      left_join_error_no_match(filter(L123.in_EJ_R_indchp_F_Yh, GCAM_region_ID == 11),
+                               by = c("fuel", "year", "sector")) %>%
+      mutate(value = value * multiplier) %>%
+      select(-multiplier, -GCAM_region_ID) ->
+      L132.in_EJ_province_indchp_F
+
+    # Apportion nation-level industrial cogen output to provinces
+    L132.in_pct_province_ind_F %>%
+      filter(fuel %in% gcam.IND_ENERGY_USE) %>%
+      mutate(sector = "chp_elec") %>%
+      # ^^ prepare for smooth join
+      left_join_error_no_match(filter(L123.out_EJ_R_indchp_F_Yh, GCAM_region_ID == 11),
+                               by = c("fuel", "year", "sector")) %>%
+      mutate(value = value * multiplier) %>%
+      select(-multiplier, -GCAM_region_ID) ->
+      L132.out_EJ_province_indchp_F
+
+
+    # PART 3: Industrial feedstocks
+    # Feedstocks are the fuel inputs to the CESY Industry -- NonEnergy plus asphalt input into construction (for roads)
+    # Split province-level data into feedstocks only, according to names of fuel inputs
+
+    L101.NBS_use_all_Mtce %>%
+      filter(EBProcess == "Construction",
+               EBMaterial == "Bitumen Asphalt") %>%
+      mutate(sector = 'industry_feedstocks', fuel = "refined liquids") ->
+      L132.in_Mtce_province_indfeed_F_unscaled.road
+
+    L101.inNBS_Mtce_province_S_F %>%
+      filter(sector == "industry_feedstocks",
+               fuel %in% unique(L1322.in_EJ_R_indfeed_F_Yh$fuel)) %>%
+      bind_rows(L132.in_Mtce_province_indfeed_F_unscaled.road[, names( L101.inNBS_Mtce_province_S_F)]) %>%
+      # First zero out NAs in years where some values are NA but not all
+      group_by(province, year) %>%
+      mutate(value = replace(value, is.na(value) & sum(value, na.rm = T) != 0, 0)) %>%
+      ungroup %>%
+      group_by(province, sector, fuel, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup ->
+      L132.in_Mtce_province_indfeed_F_unscaled
+
+    L132.in_Mtce_province_indfeed_F_unscaled %>%
+      group_by(province, sector, fuel) %>%
+      # use approx_fun rule = 2 to fill out data in years where the entire province is NA
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup %>%
+      # When there is one data point, approx_fun replaced it with NA, but what we needed is to replace all NAs with that data.
+      left_join(L132.in_Mtce_province_indfeed_F_unscaled %>% rename(org = value),
+                by = c("province", "fuel", "sector", "year")) %>%
+      group_by(sector, fuel, province) %>%
+      mutate(sum =sum(org, na.rm = T)) %>%
+      mutate(value = replace(value, is.na(value) & sum != 0, sum)) %>%
```

Comment:
I don't have a better solution. 
Combining the two lines this way may be slightly clearer: `mutate(value = replace(value, is.na(value) & sum(org[is.na(value), na.rm = T) != 0,  sum(org[is.na(value), na.rm = T))`
We can keep it for now and add comment explaining that only under the occasion when there is one data point in the group, approx_fun replaces the one value with NA. The original gcam_interp function replaces all missing values in that group with the one value when rule = 2.